source/Module/STIG/Functions.Checklist.ps1

```diff
@@ -4,128 +4,217 @@
+    
+    .EXAMPLE
+        Generate a checklist for a single STIG using DSC results obtained from Test-DscConfiguration:
+
+        $audit = Test-DscConfiguration -ComputerName localhost -ReferenceConfiguration 'C:\Dev\Utilities\SqlServerInstance_config\localhost.mof'
+        $xccdfPath = 'C:\U_MS_SQL_Server_2016_Instance_STIG_V1R7_Manual-xccdf.xml'
+        $outputPath = 'C:\SqlServerInstance_2016_V1R7_STIG_config_dscresults.ckl'
+        $ManualChecklistEntriesFile = 'C:\ManualCheckListEntriesSQL2016Instance.xml'
+        New-StigCheckList -DscResult $audit -XccdfPath $xccdfPath -OutputPath $outputPath -ManualChecklistEntries $ManualChecklistEntriesFile 
 
     .EXAMPLE
-        New-StigCheckList -ReferenceConfiguration $referenceConfiguration -XccdfPath $xccdfPath -OutputPath $outputPath
+        Generate a checklist for multiple STIGs for an endpoint using a .MOF file and a file containing STIGs to check:
+
+        $XccdfPath = Get-Content 'C:\ChecklistSTIGFiles.txt'
+        $outputPath = 'C:\SqlServer01_mof.ckl'
+        $ManualChecklistEntriesFile = 'C:\ManualCheckListEntriesSqlServer01ExcelExport.xml'
+        New-StigCheckList -DscResults $auditRehydrated -XccdfPath $XccdfPath -OutputPath $outputPath -ManualChecklistEntries $ManualChecklistEntriesFile
 
     .EXAMPLE
-        New-StigCheckList -ReferenceConfiguration $referenceConfiguration -ManualCheckFile "C:\Stig\ManualChecks\2012R2-MS-1.7.psd1" -XccdfPath $xccdfPath -OutputPath $outputPath
-        New-StigCheckList -ReferenceConfiguration $referenceConfiguration -ManualCheckFile $manualCheckFilePath -XccdfPath $xccdfPath -OutputPath $outputPath
+        Generate a checklist for multiple STIGs for an endpoint using DSC results obtained from Test-DscConfiguration, dehydrated/rehydrated using CLIXML:
+
+        $audit = Test-DscConfiguration -ComputerName localhost -MofFile 'C:\localhost.mof'
+        $audit | Export-Clixml 'C:\TestDSC.xml'
+        
+        $auditRehydrated = import-clixml C:\TestDSC.xml
+        $XccdfPath = 'C:\STIGS\SQL Server\U_MS_SQL_Server_2016_Instance_STIG_V1R7_Manual-xccdf.xml','C:\STIGS\Windows.Server.2012R2\U_MS_Windows_2012_and_2012_R2_DC_STIG_V2R19_Manual-xccdf.xml'
+        $outputPath = 'C:\SqlServer01_dsc.ckl'
+        $ManualChecklistEntriesFile = 'C:\ManualCheckListEntriesSqlServer01ExcelExport.xml'
+
+        New-StigCheckList -DscResults $auditRehydrated -XccdfPath $XccdfPath -OutputPath $outputPath -ManualChecklistEntries $ManualChecklistEntriesFile
 #>
 function New-StigCheckList
 {
     [CmdletBinding()]
-    [OutputType([xml])]
+    [OutputType([XML])]
     param
     (
         [Parameter(Mandatory = $true, ParameterSetName = 'mof')]
-        [string]
+        [Alias('MofFile')]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript(
+        {
+            If (Test-Path -Path $_ -PathType Leaf)
+            {
+                Return $True
+            } else {
```

Comment:
from a consistency perspective, all the "else" statements in this function should be structured like so:
    ...
}
else
{
    ...
}

There are other examples, this is just the first.

Reply:
Done.